src/edu/washington/escience/myria/operator/SamplingDistribution.java

```diff
@@ -0,0 +1,320 @@
+package edu.washington.escience.myria.operator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.column.Column;
+import edu.washington.escience.myria.column.builder.IntColumnBuilder;
+import edu.washington.escience.myria.column.builder.StringColumnBuilder;
+import edu.washington.escience.myria.storage.TupleBatch;
+import edu.washington.escience.myria.util.SamplingType;
+
+public class SamplingDistribution extends UnaryOperator {
+  /** Required for Java serialization. */
+  private static final long serialVersionUID = 1L;
+
+  /** The output schema. */
+  private static final Schema SCHEMA = Schema.of(ImmutableList.of(
+      Type.INT_TYPE, Type.INT_TYPE, Type.INT_TYPE, Type.STRING_TYPE),
+      ImmutableList.of("WorkerID", "StreamSize", "SampleSize", "SampleType"));
+
+  /** Total number of tuples to sample. */
+  private int sampleSize;
```

Comment:
This will be default-initialized to 0, of course, but given that one of the constructors doesn't explicitly initialize it, and it's only set much later, it might be clearer to initialize it to 0 explicitly.